Neos.Flow/Classes/Reflection/ReflectionService.php

```diff
@@ -1352,6 +1352,9 @@ public function reflectClassProperty($className, PropertyReflection $property)
             $this->classReflectionData[$className][self::DATA_CLASS_PROPERTIES][$propertyName][self::DATA_PROPERTY_ANNOTATIONS][get_class($annotation)][] = $annotation;
         }
         if (PHP_MAJOR_VERSION >= 8) {
+            if ($property->hasType() && !isset($this->classReflectionData[$className][self::DATA_CLASS_PROPERTIES][$propertyName][self::DATA_PROPERTY_TAGS_VALUES]['var'])) {
```

Comment:
This is a little bit hacky. The ReflectionService resolves the type by checking for the var annotation ([see](https://github.com/neos/flow-development-collection/blob/5abe1256ffaf79848a8c6ebe0df31bb4389f0547/Neos.Flow/Classes/Reflection/ReflectionService.php#L1688-L1699)). This was the only way I found to not refactor more code as needed. There are several other parts in the core that use the same way to resolve the type of the property.